metadata/src/main/java/edu/unc/lib/dl/util/PremisEventBuilder.java

```diff
@@ -0,0 +1,93 @@
+package edu.unc.lib.dl.util;
+
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Resource;
+
+import edu.unc.lib.dl.event.PremisLogger;
+import edu.unc.lib.dl.rdf.Premis;
+
+public class PremisEventBuilder {
+	private String eventId;
+	private Resource eventType;
+	private Date date;
+	private Model model;
+	private PremisLogger premisLogger;
+	private Resource premisObjResc;
+	
+	public PremisEventBuilder(String eventId, Resource eventType, Date date,
+			PremisLogger premisLogger) {
+		this.eventId = eventId;
+		this.eventType = eventType;
+		this.date = date;
+		this.premisLogger = premisLogger;
+		addEvent();
+	}
+
+	private PremisEventBuilder addEvent() {
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+		Resource premisObjResc = getResource();
+
+		premisObjResc.addProperty(Premis.hasEventType, this.eventType);
+		premisObjResc.addProperty(Premis.hasEventDateTime, dateFormat.format(this.date));
+		
+		return this;
+	}
+	
+	public PremisEventBuilder addEventDetail(String message, Object... args) {
+		if(args != null) {
+			message = MessageFormat.format(message, args);
+		}
+		Resource premisObjResc = getResource();
+		premisObjResc.addProperty(Premis.hasEventDetail, message);
+		
+		return this;
+	}
+	
+	public PremisEventBuilder addSoftwareAgent(String name, String versionNumber) {	
+		Resource premisObjResc = getResource();
+		
+		premisObjResc.addProperty(Premis.hasAgentName, name+" ("+versionNumber+")");
+		premisObjResc.addProperty(Premis.hasAgentType, "Software");
+
+		return this;
+	}
+	
+	public PremisEventBuilder addDerivitave(String sourceDataStream, String destDataStream) {
+		Resource premisObjResc = getResource();
+		
+		premisObjResc.addProperty(Premis.hasAgentName, sourceDataStream);
+		premisObjResc.addProperty(Premis.hasAgentType, "Source Data");
+		premisObjResc.addProperty(Premis.hasAgentName, destDataStream);
+		premisObjResc.addProperty(Premis.hasAgentType, "Derived Data");
+		
+		return this;
+	}
+	
+	public Model create() {
+		return getModel();
```

Comment:
Since the event is really a resource rather than a model, it would make sense to me for this to return the event resource instead.  You can get back to the model in other places if needed by calling .getModel() on the resource.